src/reversion_rest_framework/mixins.py

```diff
@@ -14,24 +18,48 @@ class BaseHistoryModelMixin:
 
 
 class HistoryOnlyMixin(BaseHistoryModelMixin):
+
+    def build_serializer(self, instance_class: type, queryset: QuerySet, many: Optional[bool] = False):
+        """wraps the original serializer within the Version serializer on the field_dict field
+        """
+
+        class _InstanceSerializer(ModelSerializer):
+
+            class Meta:
+                model = instance_class
+                fields = "__all__"
+
+        class _VersionsSerializer(self.version_serializer):
+            field_dict = SerializerMethodField()
+
+            @staticmethod
+            def get_field_dict(obj):
+                model_serializer = _InstanceSerializer(data=obj.field_dict)
+                model_serializer.is_valid(raise_exception=True)
+                original_serializer = self.get_serializer(model_serializer.validated_data)
+                return original_serializer.data
+
+        return _VersionsSerializer(queryset, many=many)
+
     @action(detail=True, methods=['GET'], name='Get History')
     def history(self, request, pk=None):
         instance = self.get_object()
         versions = Version.objects.get_for_object(instance).order_by(
             '-revision__date_created'
         )
         page = self.paginate_queryset(versions)
+
         if page is not None:
-            serializer = self.version_serializer(page, many=True)
+            serializer = self.build_serializer(instance.__class__, page, many=True)
             return self.get_paginated_response(serializer.data)
-        serializer = self.version_serializer(versions, many=True)
+        serializer = self.build_serializer(instance.__class__, versions, many=True)
         return Response(serializer.data)
 
     @action(detail=True, methods=['GET'], name='Get Historic Version', url_path=r'history/(?P<version_pk>\d+)')
     def version(self, request, pk=None, version_pk=None):
         instance = self.get_object()
         version = get_object_or_404(Version.objects.get_for_object(instance), id=version_pk)
-        serializer = self.version_serializer(version)
+        serializer = self.build_serializer(instance.__class__, version)
```

Comment:
Can we use `build_serializer` also in `deleted` and `revert` actions?

Reply:
good point, added in d3807c0
I am wondering if `@deleted` should be paginated as well?

Reply:
> I am wondering if @deleted should be paginated as well?

it might be a good idea, let me work on that 😉